src/posix/platform/netif.cpp

```diff
@@ -851,28 +851,31 @@ static void processTransmit(otInstance *aInstance)
     }
 }
 
-static void logAddrEvent(bool isAdd, bool isUnicast, struct sockaddr_in6 &addr6, otError error)
+static void logAddrEvent(bool isAdd, const ot::Ip6::Address &aAddress, otError error)
 {
     char addressString[INET6_ADDRSTRLEN + 1];
 
     // these parameters may not be used if logging is disabled at compile time
-    OT_UNUSED_VARIABLE(isUnicast);
-    OT_UNUSED_VARIABLE(addr6);
+    OT_UNUSED_VARIABLE(aAddress);
     OT_UNUSED_VARIABLE(addressString);
 
-    if ((error == OT_ERROR_NONE) || ((isAdd) && (error == OT_ERROR_ALREADY)) ||
+    if ((error == OT_ERROR_NONE) || ((isAdd) && (error == OT_ERROR_ALREADY || error == OT_ERROR_REJECTED)) ||
         ((!isAdd) && (error == OT_ERROR_NOT_FOUND)))
     {
-        otLogNotePlat("[netif] %s [%s] %s%s", isAdd ? "ADD" : "DEL", isUnicast ? "U" : "M",
-                      inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
-                      error == OT_ERROR_ALREADY ? " (already subscribed, ignored)"
-                                                : error == OT_ERROR_NOT_FOUND ? " (not found, ignored)" : "");
+        otIp6AddressToString(&aAddress, addressString, sizeof(addressString));
```

Comment:
may directly use `ToString()` method of class `ot::Ip6::Address`, or should the formal parameter be of type`otIp6Address`